tests/test_custom_learner.py

```diff
@@ -70,6 +77,9 @@ def tearDown():
                         glob(join(output_dir, 'test_model_custom_learner_*'))):
         os.unlink(output_file)
 
+    for log_file in glob(join(log_dir, '*')):
```

Comment:
I think rather than deleting all log files which could affect other tests, we should only delete the specific log file we are actually generating.

Reply:
Done! I've also done a little bit of cleaning up to the tearDown function and instead of creating a new log dir, I've written the log file to an `output` dir, since the log files are being written to the `output` dir in a few other places.

Reply:
Thanks for doing this. One minor nit I want to pick is that I am not a fan of single letter variable names and nested list comprehensions when they aren't necessary and sacrifice readability. I think in this case, nested for loops would work just fine.

Reply:
I totally get that - this practice of mine has always been a little polarizing :P 
Made this change.

Reply:
Thank you!